enterprise/backend/src/metabase_enterprise/caching/api.clj

```diff
@@ -0,0 +1,93 @@
+(ns metabase-enterprise.caching.api
+  (:require
+   [compojure.core :refer [GET]]
+   [java-time.api :as t]
+   [metabase.api.common :as api]
+   [metabase.api.common.validation :as validation]
+   [metabase.api.routes.common :refer [+auth]]
+   [metabase.util.cron :as u.cron]
+   [metabase.util.i18n :refer [tru]]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+(api/defendpoint GET "/"
+   "Return cache configuration."
+  [:as {{:strs [model collection]
+         :or   {model "root"}}
+        :query-params}]
+  {model      [:set {:decode/string (fn [x] (cond (set? x)    x
+                                                  (vector? x) (set x)
+                                                  (some? x)   #{x}))}
+               [:enum "root" "database" "collection" "dashboard" "question"]]
+   collection [:maybe ms/PositiveInt]}
+  (validation/check-has-application-permission :setting)
+
+  (let [model (cond-> model
+                (some #{"dashboard" "question"} model) (conj "collection"))
+        opts  [:model [:in model]
+               :collection_id collection]
+        items (apply t2/select :model/CacheConfig opts)]
+    {:items items}))
+
+(api/defendpoint PUT "/"
+  "Store cache configuration."
+  [:as {{:keys [model model_id strategy]} :body}]
+  {model    [:enum "root" "database" "collection" "dashboard" "question"]
+   model_id ms/IntGreaterThanOrEqualToZero
+   strategy [:and
+             [:map
+              [:type [:enum :nocache :ttl :duration :schedule :query]]]
+             [:multi {:dispatch :type}
+              [:nocache  [:map
+                          [:type keyword?]]]
+              [:ttl      [:map
+                          [:type keyword?]
+                          [:multiplier ms/PositiveInt]
+                          [:min_duration ms/PositiveInt]]]
+              [:duration [:map
+                          [:type keyword?]
+                          [:duration ms/PositiveInt]
+                          [:unit [:enum "hours" "minutes" "seconds" "days"]]]]
+              [:schedule [:map
+                          [:type keyword?]
+                          [:schedule u.cron/CronScheduleString]]]
+              [:query    [:map
+                          [:type keyword?]
+                          [:field_id int?]
+                          [:aggregation [:enum "max" "count"]]
+                          [:schedule string?]]]]]}
+  (validation/check-has-application-permission :setting)
+  (when (and (= model "root") (not= model_id 0))
+    (throw (ex-info (tru "Root configuration is only valid with model_id = 0") {:status-code 400
+                                                                                :model_id    model_id})))
```

Comment:
this could be a helper so we can re-use it in DELETE endpoint

Reply:
done in 368c44122cafcbd6654eec45831490833445b341